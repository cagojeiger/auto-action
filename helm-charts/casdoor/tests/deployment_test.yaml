suite: test deployment
templates:
  - deployment.yaml
chart:
  version: 1.0.0
  appVersion: 1.2.3
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}
  - it: has extraVolume and extraVolumeMounts
    set:
      extraVolumes:
        - name: extra-volume
          emptyDir: {}
      extraVolumeMounts:
        - name: extra-volume
          mountPath: /extra-volume
    asserts:
      - matchSnapshot: {}
