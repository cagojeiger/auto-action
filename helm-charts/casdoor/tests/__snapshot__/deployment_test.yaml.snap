has extraVolume and extraVolumeMounts:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor-helm-charts
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: casdoor-helm-charts-1.0.0
      name: RELEASE-NAME-casdoor-helm-charts
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor-helm-charts
      template:
        metadata:
          annotations:
            checksum/config: 529bc7ea51d30d00fefc46d24be7446d0637939827bccc1c3f03755f70059470
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor-helm-charts
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:1.2.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
                - mountPath: /extra-volume
                  name: extra-volume
          securityContext: {}
          serviceAccountName: RELEASE-NAME-casdoor-helm-charts
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-helm-charts-config
            - emptyDir: {}
              name: extra-volume
manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor-helm-charts
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: casdoor-helm-charts-1.0.0
      name: RELEASE-NAME-casdoor-helm-charts
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: casdoor-helm-charts
      template:
        metadata:
          annotations:
            checksum/config: 529bc7ea51d30d00fefc46d24be7446d0637939827bccc1c3f03755f70059470
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: casdoor-helm-charts
        spec:
          containers:
            - env:
                - name: RUNNING_IN_DOCKER
                  value: "true"
              image: casbin/casdoor:1.2.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: casdoor-helm-charts
              ports:
                - containerPort: 8000
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /conf
                  name: config-volume
          securityContext: {}
          serviceAccountName: RELEASE-NAME-casdoor-helm-charts
          volumes:
            - name: config-volume
              projected:
                defaultMode: 420
                sources:
                  - configMap:
                      items:
                        - key: app.conf
                          path: app.conf
                      name: RELEASE-NAME-casdoor-helm-charts-config
